cmd/minishift/cmd/config/validations.go

```diff
@@ -38,7 +38,9 @@ func IsValidDriver(string, driver string) error {
 }
 
 func RequiresRestartMsg(string, string) error {
-	fmt.Fprintln(os.Stdout, "To apply the changes, you must delete the current VM with `minishift delete` and start a new VM with `minishift start`.")
+	fmt.Fprintln(os.Stdout, "To apply the changes, "+
+		"you must delete the current VM with `minishift delete` if VM exist "+
```

Comment:
I think we can do better than this. We literally should check whether the machine exists. If so we write print:

"No Minishift VM defined. Settings will be applied on next 'minishift start'"

If the machine does exist:

"Settings are only applied when a new Minishift instance is created. You currently have an existing VM. To apply the changes, you must delete the current VM with `minishift delete` and start a new VM with `minishift start`."